diff --git a/Makefile b/Makefile
index 5ea21183..7410c45b 100644
--- a/Makefile
+++ b/Makefile
@@ -279,7 +279,7 @@ endif
 
 # convert "include/tccdefs.h" to "tccdefs_.h"
 %_.h : include/%.h conftest.c
-	$S$(CC) -DC2STR $(filter %.c,$^) -o c2str.exe && ./c2str.exe $< $@
+	$S$(CC) -DC2STR $(filter %.c,$^) -o c2str.exe && $$outdir/lib/libc.so ./c2str.exe $< $@
 
 # target specific object rule
 $(X)%.o : %.c $(LIBTCC_INC)
diff --git a/lib/Makefile b/lib/Makefile
index 4727beeb..060eb346 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -11,12 +11,12 @@ XCFG = $(or $(findstring -win,$T),-unx)
 S = $(if $(findstring yes,$(SILENT)),@$(info * $@))
 
 TCC = $(TOP)/$(X)tcc$(EXESUF)
-XTCC ?= $(TOP)/$(X)tcc$(EXESUF)
+XTCC ?= $$outdir/lib/libc.so $(TOP)/$(X)tcc$(EXESUF)
 XCC = $(XTCC)
 XAR = $(XTCC) -ar
 XFLAGS-unx = -B$(TOPSRC)
 XFLAGS-win = -B$(TOPSRC)/win32 -I$(TOPSRC)/include
-XFLAGS = $(XFLAGS$(XCFG)) -I$(TOP)
+XFLAGS = $(XFLAGS$(XCFG)) -I$(TOP) -I $$outdir/include
 BFLAGS = -bt
 
 # in order to use gcc, type: make <target>-libtcc1-usegcc=yes
